nifi-docs/src/main/asciidoc/walkthroughs.adoc

```diff
@@ -0,0 +1,1099 @@
+
+|=======================================================================================================================
+|Description        | Instructions on configuring and securing a 3 node cluster
+|Purpose            | A NiFi cluster can perform flow operations in parallel and provide resiliency if nodes are unavailable.
+|Starting Point     | <<installing-apache-nifi,Installing Apache NiFi>> or <<building-nifi-from-source,Building NiFi from Source>>
+|Expected Outcome   | 3 nodes of latest version of Apache NiFi in secured cluster running on local machine
+|Helpful Reading    | <<securing-nifi-with-tls-toolkit,Securing NiFi with TLS Toolkit>>, <<creating-a-nifi-cluster,Creating a NiFi Cluster>>
+|Estimated Duration | 15 minutes
+|=======================================================================================================================
+
+Similar to a standalone instance, a NiFi cluster must have TLS configured to enable authentication and authorization mechanisms. However, the cluster communication protocol used by NiFi and the framework authentication and authorization must also be configured to allow nodes to communicate.
+
+For this guide, the three nodes will be referred to as `node1`, `node2`, and `node3`. Each will have a `$NIFI_HOME` environment variable which points to the respective "NiFi home directory".
+
+Prerequisites:
+
+* DNS configuration for three hosts using the addresses `node1.nifi`, `node2.nifi`, `node3.nifi`
+** This can be all side-by-side on the same physical host, using virtual machines, or using independent physical machines
+* Ports open to allow the nodes to communicate
+** If these machines are subject to a firewall, ensure that the following ports are open:
+*** `2181`, `2888`, `3888` -- ZooKeeper ports
+*** `6342` -- Load-balancing port
+*** `9443` -- HTTPS port for NiFi UI/API
+*** `10443` -- Site to site port
+*** `11443` -- Cluster communications port
+** If all the nodes will run on the same machine, the port scheme will end in the respective node identifier to avoid conflicts (e.g. HTTPS port `9441` for `node1`, `9442` for `node2`, `9443` for `node3`). Ensure that the following ports are open:
+*** `2181` - `2183` -- ZooKeeper ports
+*** `2881` - `2883` -- ZooKeeper ports
+*** `3881` - `3883` -- ZooKeeper ports
+*** `6341` - `6343` -- Load-balancing port
+*** `9441` - `9443` -- HTTPS port for NiFi UI/API
+*** `10441` - `10443` -- Site to site port
+*** `11441` - `11443` -- Cluster communications port
+
+The end result will consist of a self-signed external CA (the root), a keystore and truststore containing the necessary certificates for each NiFi node to operate, and a client certificate for a human user to access NiFi.
+
+image::nifi-cluster-tls-toolkit-certificate-diagram.png["NiFi Cluster with TLS Toolkit Certificates Diagram"]
```

Comment:
Don't have to fix for this version, but some of the text is difficult to read in this image.